app/config/settings.py

```diff
@@ -419,6 +421,13 @@
     "django.contrib.flatpages.middleware.FlatpageFallbackMiddleware",
     # Redirects last as they're a last resort
     "django.contrib.redirects.middleware.RedirectFallbackMiddleware",
+    # 2FA middleware, needs to be after subdomina, flatpage and redirect middleware
+    # TwoFactorMiddleware resets the login flow if another page is loaded
+    # between login and successfully entering two-factor credentials. We're subsetting
+    # the original allauth_2fa middleware to pass the correct urlconf.
+    "grandchallenge.core.middleware.TwoFactorMiddleware",
+    # Force 2AF for staff users
+    "grandchallenge.core.middleware.RequireStaffAndSuperuser2FAMiddleware",
```

Comment:
There are a few typos in this, 2AF should be 2FA and subdomina should be subdomain.

Why do they need to be after FlatpageFallbackMiddleware and RedirectFallbackMiddleware? Does this affect our flatpages and redirects, especially on sub-domains?

Reply:
What tests would break if these were out of order? Or not after the subdomain middleware?

Reply:
The Flatpages redirect tests fail if I put the middleware before the flatpage middleware. With the current order of the middleware everything works as intended, but I will take another, closer look at the 2FA middleware (also add a test for it).

Reply:
Yes I think that makes sense.